tools/pre_build_script.py

```diff
@@ -59,6 +60,15 @@
 
 def generate_version_file(version_git_dir):
 
+    # Get the git description.
+    print("#> Getting git description")
+    try:
+        git_description = check_output("git describe HEAD", shell=True)
```

Comment:
My intent is to have separate CI tasks for testing and release, with tagging occurring before the release build. The release builds would be reproducible (and in most case other builds as well). @flit suggested we use this instead of a short hash, at first I was not really sure of the value, but now it seems to me the value is outside of releases, where you get some sense of the distance to the last tag.

Reply:
So with this change would the release process consist of running builds twice, first to test the release candidate, and next to test the tagged release? Also would this cause problems if you tagged the same commit twice (For example in mbed-os mbed-os-6.14.0 and mbed-os-6.14.0-rc1 are on the same commit)? As for getting a description of were a build was done you could run  `git describe <SHA from details.txt>` from a cloned DAPLink repo which should provide the same information. 

I'm not convinced the benefits of using `git describe HEAD` outweigh the downsides.